### PR TITLE
Teacher UI in progress redesign

### PR DESCRIPTION
**Pull request by Bjvanminnen**
image

### CODE REVIEW

File: apps/src/templates/progress/ProgressLesson.jsx [hunk 1]
```diff
@@ -8,16 +8,29 @@ import { ViewType } from '@cdo/apps/code-studio/stageLockRedux';
...
     background: color.lightest_gray,
     borderWidth: 1,
     borderColor: color.border_gray,
     borderStyle: 'solid',
     borderRadius: 2,
+  },
+  rightCol: {
+    display: 'table-cell',
+    verticalAlign: 'top',
+    // TODO - fixed width will likely get us into trouble for other languages
```
Comment (Bjvanminnen) [inline apps/src/templates/progress/ProgressLesson.jsx]: think about this some more brent

File: apps/src/templates/progress/StageLock.jsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
+/**
```
Comment (Bjvanminnen) [inline apps/src/templates/progress/StageLock.jsx:1]: Potentially contentious decision here and with HiddenStageToggle to name a component identically to the component it is replacing (the difference being where they live).

File: apps/src/redux.js [hunk 3]
```diff
@@ -61,19 +61,32 @@ if (IN_UNIT_TEST) {
   };
 }
 
+if (IN_STORYBOOK) {
+  // Storybooks need the ability to create multiple distinct stores instead of
+  // using a singletone
```
Comment (islemaster) [inline apps/src/redux.js]: Singletone :notes: or singleton?

File: apps/src/redux.js [hunk 4]
```diff
@@ -61,19 +61,32 @@ if (IN_UNIT_TEST) {
...
 
   return reduxStore;
 }
 
 /**
+ * Create our store
+ */
+function createStoreWithReducers() {
+  return createStore(Object.keys(globalReducers).length > 0 ?
+    redux.combineReducers(globalReducers) : s => s);
```
Comment (islemaster) [inline apps/src/redux.js:86]: Weird... how is line 84 covered but line 85 is not?

File: apps/webpack.js [hunk 5]
```diff
@@ -122,6 +122,7 @@ var storybookConfig = _.extend({}, baseConfig, {
     new webpack.ProvidePlugin({React: 'react'}),
     new webpack.DefinePlugin({
       IN_UNIT_TEST: JSON.stringify(false),
+      IN_STORYBOOK: JSON.stringify(true),
```
Comment (islemaster) [inline apps/webpack.js:125]: I'm surprised storybook doesn't already provide this somehow, but I looked around and it doesn't seem to. :+1:

File: apps/src/templates/progress/HiddenStageToggle.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    borderTopRightRadius: 0,
+    borderBottomRightRadius: 0,
+  },
+  rightButton: {
+    borderTopLeftRadius: 0,
+    borderBottomLeftRadius: 0,
+  },
+};
+
+const HiddenStageToggle = React.createClass({
```
Comment (islemaster) [inline apps/src/templates/progress/HiddenStageToggle.js:29]: Might not be worth it yet, but I bet extracting a generic toggle component in this new style will be useful down the road.

**Pull request merged**